resource-dispatcher/execution/plugins/git.py

```diff
@@ -14,60 +14,64 @@ def run_action(params):
     branch = params["branch"] if "branch" in params else "master"
     message = params["message"] if "message" in params else "Commit message"
     author_name = params["author_name"] if "author_name" in params else "Automated Tool"
-    author_email = params["author_email"] if "author_email" in params else "email@email.com"
+    author_email = (
+        params["author_email"] if "author_email" in params else "email@email.com"
+    )
     url = params["url"] if "url" in params else None
 
     if "secret" in params and url and not url.startswith("http"):
-        env = {
-            "GIT_SSH_COMMAND": generate_ssh_command(params)
-        }
+        env = {"GIT_SSH_COMMAND": generate_ssh_command(params)}
     else:
         env = {}
 
     if params["action"] == "clone" or params["action"] == "pull":
-        repo = fetch(url, repo_directory, branch, env)
+        fetch(url, repo_directory, branch, env)
     elif params["action"] == "add-all-changes":
-        repo = add_all(repo_directory)
+        add_all(url, repo_directory)
     elif params["action"] == "commit":
-        commit(repo_directory, message, author_name, author_email)
+        commit(url, repo_directory, message, author_name, author_email)
     elif params["action"] == "push":
         push(url, repo_directory, env)
 
 
 def fetch(url, repo_directory, branch, env={}):
     if not os.path.exists(repo_directory):
         os.makedirs(repo_directory)
-        print(f"Cloning {url}...")
-        repo = Repo.clone_from(url, repo_directory, branch=branch, env=env)
-        print(f"Cloned {url}")
+        try:
+            print(f"Cloning {url}...")
+            Repo.clone_from(url, repo_directory, branch=branch, env=env)
+            print(f"Cloned {url}")
+        except:
+            pass
```

Comment:
Should something be logged here? Curious if there's a reason to ignore a clone issue silently or if we should log it out like in lines 76-77. Otherwise everything looks good to me 👍

Reply:
Yes, I agree. I will capture the error and log/handle it better.

The main issue I am seeing is when an empty repository is in the Gitlab group, it will break the loop. Passing on the `fetch` stage works but then the add => commit => push flow keeps running with that item introducing more errors. It might be an added bonus to pop the failed item out of the list so that later steps in the task stop doing work on the empty repo.

I'll get the logging in first then see what is possible with the latter.

Reply:
Since dispatcher is currently assuming that origin/master exists, this will help identify any empty repositories.

```
All files added: ssh://git@gitlab.example.com/project/engagement.git
Nothing to commit: ssh://git@gitlab.example.com/project/engagement.git
Pushed repository: ssh://git@gitlab.example.com/project/engagement.git
Remote origin does not exist: ssh://git@gitlab.example.com/project/engagement.git```

Reply:
👍 👍 👍